compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/lir/aarch64/AArch64ArrayFillOp.java

```diff
@@ -0,0 +1,333 @@
+import static jdk.graal.compiler.asm.aarch64.AArch64Address.AddressingMode.IMMEDIATE_POST_INDEXED;
+import static jdk.graal.compiler.lir.LIRInstruction.OperandFlag.REG;
+import static jdk.vm.ci.aarch64.AArch64.r10;
+import static jdk.vm.ci.aarch64.AArch64.r5;
+import static jdk.vm.ci.aarch64.AArch64.r6;
+import static jdk.vm.ci.aarch64.AArch64.r7;
+import static jdk.vm.ci.aarch64.AArch64.r8;
+import static jdk.vm.ci.aarch64.AArch64.r9;
+import static jdk.vm.ci.code.ValueUtil.asRegister;
+
+import jdk.graal.compiler.asm.Label;
+import jdk.graal.compiler.asm.aarch64.AArch64Address;
+import jdk.graal.compiler.asm.aarch64.AArch64Assembler.ConditionFlag;
+import jdk.graal.compiler.asm.aarch64.AArch64Assembler.ShiftType;
+import jdk.graal.compiler.asm.aarch64.AArch64MacroAssembler;
+import jdk.graal.compiler.debug.GraalError;
+import jdk.graal.compiler.lir.LIRInstructionClass;
+import jdk.graal.compiler.lir.Opcode;
+import jdk.graal.compiler.lir.asm.CompilationResultBuilder;
+import jdk.vm.ci.aarch64.AArch64Kind;
+import jdk.vm.ci.code.Register;
+import jdk.vm.ci.meta.JavaKind;
+import jdk.vm.ci.meta.Value;
+
+/**
+ * Emits code which fills an array with a constant value. The assembly code in this intrinsic was
+ * based in the HotSpot's version of the same intrinsic.
+ */
+@Opcode("ARRAYS_FILL")
+public final class AArch64ArrayFillOp extends AArch64ComplexVectorOp {
+    public static final LIRInstructionClass<AArch64ArrayFillOp> TYPE = LIRInstructionClass.create(AArch64ArrayFillOp.class);
+
+    private JavaKind elementType;
+    @Alive({REG}) protected Value array;
+    @Alive({REG}) protected Value arrayBaseOffset;
+    @Alive({REG}) protected Value length;
+    @Alive({REG}) protected Value value;
+    @Temp protected Value[] temps;
+
+    public AArch64ArrayFillOp(JavaKind kind, Value array, Value arrayBaseOffset, Value length, Value value) {
+        super(TYPE);
+
+        GraalError.guarantee(array.getPlatformKind() == AArch64Kind.QWORD, "pointer value expected");
+        GraalError.guarantee(length.getPlatformKind() == AArch64Kind.DWORD, "integer value expected in 'length'");
+        GraalError.guarantee(value.getPlatformKind() == AArch64Kind.DWORD, "integer value expected in 'value'.");
+
+        this.elementType = kind;
+        this.array = array;
+        this.arrayBaseOffset = arrayBaseOffset;
+        this.length = length;
+        this.value = value;
+
+        this.temps = new Value[]{
+                        r5.asValue(),
+                        r6.asValue(),
+                        r7.asValue(),
+                        r8.asValue(),
+                        r9.asValue(),
+                        r10.asValue()};
+    }
+
+    @Override
+    @SuppressWarnings("fallthrough")
+    public void emitCode(CompilationResultBuilder crb, AArch64MacroAssembler masm) {
+        int shift = -1;
+        int operationWidthB = 8;
+        int operationWidthH = 16;
+        int operationWidthW = 32;
+        int operationWidthDW = 64;
```

Comment:
Our normal style is to have these magic operation size numbers as explicit constants in the code. The difference between `32` and `64` is just much easier to see than the difference between `operationWidthW` and `operationWidthDW`.

Also note that you're defining `DW` as 64 bits, but that doesn't match our conventions: `AArch64Kind.DWORD` is 32 bits. So please drop these symbolic constants.